### PR TITLE
Zip location fixes, zipping, parallel uploads

### PR DESCRIPTION
**Pull request by rukkanen**

### CODE REVIEW

**Comment by rukkanen** [comment]
@mattisii pingpong

File: src/main/java/com/synopsys/protecode/sc/jenkins/ProtecodeScPlugin.java [hunk 1]
```diff
@@ -512,7 +352,7 @@ public boolean configure(StaplerRequest req, JSONObject formData)
 
     public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item context) {
       // TODO Find a nice way to use this to fetch possible groups
-      //  - this might be impossible in this scope
+      //  - this might be impossible in this sc String jobNameope
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/ProtecodeScPlugin.java]: typo

File: src/main/java/com/synopsys/protecode/sc/jenkins/Scanner.java [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+      if (zipName.isPresent()) {
+        if (files.size() != 1) {
+          // For some obscure cases where the zip could not be made, but no exception was raised.
+          zippingInUse = false;
+        }
+      }
+      sendFiles(files, zipName);
+      
+    } else {
+      LOGGER.log(Level.WARNING, "Gettgin from URL");
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/Scanner.java]: typo

File: src/main/java/com/synopsys/protecode/sc/jenkins/Scanner.java [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+            addUploadResponse(log, directoryToScan, result, NO_ERROR);
+          }
+        } 
+      );
+    }
+    
+    try {
+      waitForUploadResponses(files.size(), log);
+    } catch (NullPointerException e) {
+
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/Scanner.java]: Do something for this, log atleast?

File: src/main/java/com/synopsys/protecode/sc/jenkins/types/BuildVerdict.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * A collect all class for protecode scan result/build status
+ * @author rukkanen
+ */
+public @Data class BuildVerdict {
+  private final boolean failOnVulns;
+ 
+  private boolean zippingUsed = false;
+  private int filesFound = 0;
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/types/BuildVerdict.java]: need a long?

File: src/main/java/com/synopsys/protecode/sc/jenkins/types/BuildVerdict.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  private boolean filesWithUntriaVulns = false;
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/types/BuildVerdict.java]: Untriaged

File: src/main/java/com/synopsys/protecode/sc/jenkins/utils/UtilitiesFile.java [hunk 6]
```diff
@@ -65,9 +65,9 @@ private UtilitiesFile() {
    * @return List of FilePaths produced as artifacts
    */
   // TODO: CLEAN! And add option to use (w/ option to scan artifacts)
-  public static List<FilePath> getArtifacts(Run<?, ?> run) {
-    return getArtifacts(run, ALL_FILES_PATTERN);
-  }
+  //public static List<FilePath> getArtifacts(Run<?, ?> run) {
```
Comment (bd-matti) [inline src/main/java/com/synopsys/protecode/sc/jenkins/utils/UtilitiesFile.java]: Commented code

**Pull request merged**